src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -255,7 +256,7 @@ internal static int CombineHashCodes(int h1, int h2, int h3)
 
         internal static int CombineHashCodes(int h1, int h2, int h3, int h4)
         {
-            return CombineHashCodes(CombineHashCodes(h1, h2), CombineHashCodes(h3, h4));
+            return CombineHashCodes(CombineHashCodes(h1, h2, h3), h4);
```

Comment:
Why did you change these calls?  The previous version had three calls to CombineHashCodes total.  This one has four: two in this method, and then two in the CombineHashCodes(int, int, int) overload.  Same question about the other changes below.  What's the benefit of these changes?